components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4569,4 +4570,40 @@ public static boolean isFederatedRoleBasedAuthzEnabled(String clientId) throws I
         }
         return isFederatedRoleBasedAuthzEnabled;
     }
+
+    public static String[] getRequestedOIDCScopes(String[] requestedScopes)
+            throws IdentityOAuthAdminException {
+        if (requestedScopes == null) {
+            throw new IdentityOAuthAdminException("Requested scopes are null");
```

Comment:
If requested scopes is null, shouldn't we just return an empty array?
Does the callee expect an exception to be thrown?